### PR TITLE
Enable `block` variable in view

### LINKED ISSUE #27302: Variable `block` can't be used in view
### Steps to reproduce

Since PR #26672 has been merged, variable `block` in partial view doesn't work, which previously worked.

In erb file:

```erb
<%= render @blocks >
```

`app/views/blocks/_block.html.erb`

```erb
<%= block %>
```

This raises following error:

```
ActionView::Template::Error: undefined local variable or method `block' for #<ActionView::Base:0x00xxxxx>
```

### Expected behavior

`block` variable can be used in view. (can be used In Rails v5.0.0)

### Actual behavior

`block` variable raises `ActionView::Template::Error`.

### System configuration

**Rails version**:

Latest version.

**Ruby version**:

ruby 2.3.3

### PR DESCRIPTION
**Pull request by toshimaru**
### Summary

Enabling `block` variable in view.

This fixes #27302

### CODE REVIEW

**Comment by matthewd** [comment]
Instead of excluding the one value here, let's split `DELEGATION_RESERVED_METHOD_NAMES` in half, between the real keywords (most of them), and the couple (including `block`) that are just reserved in `delegate` because they're used in the implementation.

File: activesupport/lib/active_support/core_ext/module/delegation.rb [hunk 1]
```diff
@@ -6,11 +6,12 @@ class Module
   # option is not used.
   class DelegationError < NoMethodError; end
 
+  RUBY_RESERVED_KEYWORDS = %w(alias and BEGIN begin break case class def defined? do
+  else elsif END end ensure false for if in module next nil not or redo rescue retry
+  return self super then true undef unless until when while yield)
+  DELEGATION_RESERVED_KEYWORDS = %w(_ arg args block)
```
Comment (toshimaru) [inline activesupport/lib/active_support/core_ext/module/delegation.rb]: I extracted only delegation reserved keywords like this:

```
$ ruby -v
ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin15]

$ cat extract_keywords.rb
original_keywords = %w(_ arg args alias and BEGIN begin block break case class def defined? do else elsif END end ensure false for if in module next nil not or redo rescue retry return self super then true undef unless until when while yield)
keywords = []

original_keywords.each do |keyword|
  begin
    eval("#{keyword} = 1")
  rescue SyntaxError => e
    keywords << keyword
  end
end

p keywords
p original_keywords - keywords

$ ruby extract_keywords.rb
["alias", "and", "BEGIN", "begin", "break", "case", "class", "def", "defined?", "do", "else", "elsif", "END", "end", "ensure", "false", "for", "if", "in", "module", "next", "nil", "not", "or", "redo", "rescue", "retry", "return", "self", "super", "then", "true", "undef", "unless", "until", "when", "while", "yield"]
["_", "arg", "args", "block"]
```

File: activesupport/lib/active_support/core_ext/module/delegation.rb [hunk 2]
```diff
@@ -6,11 +6,12 @@ class Module
   # option is not used.
   class DelegationError < NoMethodError; end
 
+  RUBY_RESERVED_KEYWORDS = %w(alias and BEGIN begin break case class def defined? do
```
Comment (toshimaru) [inline activesupport/lib/active_support/core_ext/module/delegation.rb]: Not sure the constant naming and defined location are good. 

@matthewd What do you think?

**Comment by matthewd** [comment]
96f0df1914db51b9dc5ca5d2414748185f4f5b1e

**Pull request merged**